src/GitTfs/Core/TfsInterop/TfsPlugin.cs

```diff
@@ -161,6 +161,9 @@ public virtual void Initialize(IAssemblyScanner scan)
 
         public virtual void Initialize(ConfigurationExpression config)
         {
+            // Mark the ITfsHelper as a singleton to ensure that we create it only once.
+            // Otherwise, it is created e.g. for every remote which is wasteful.
+            config.For<ITfsHelper>().Singleton();
```

Comment:
Other singletons are marked with `[StructureMapSingleton]`. Can the plugins be marked that way instead?

Reply:
That is exactly what I tried first, but the current code only looks at that attribute of the assembly where the Program.cs is loated in, see
https://github.com/git-tfs/git-tfs/blob/1123ad84a8064540bf98faa01a8d1dfeae97d9cb/src/GitTfs/Program.cs#L178
It doesn't look at code implemented in other assemblies.

AFAIK the usage of the attribute is only for code where we actually also have one real implementation, e.g. like the [Globals](https://github.com/git-tfs/git-tfs/blob/master/src/GitTfs/Globals.cs#L13).
But for the `ITfsHelper`, we actually have multiple implementations, so I would prefere to keep it as is.

Reply:
Ah, right. 👍